src/BoundsInference.cpp

```diff
@@ -259,11 +272,61 @@ class BoundsInference : public IRMutator {
             return true;
         }
 
+        // Determine if the current producing stage are fused with other
+        // stage (i.e. the consumer stage) at 'var'.
+        bool is_fused_with_others(const vector<vector<Function>> &fused_groups,
+                                  const vector<set<FusedPair>> &fused_pairs,
+                                  const Function &producing_func, int producing_stage,
+                                  string var, string consumer_name, int consumer_stage) {
+            // Find the fused group this producing stage belongs to.
+            int index = -1;
+            for (size_t i = 0; i < fused_groups.size(); ++i) {
+                const auto iter = std::find_if(fused_groups[i].begin(), fused_groups[i].end(),
+                    [&producing_func](const Function& f) { return (f.name() == producing_func.name()); });
+                if (iter != fused_groups[i].end()) {
+                    index = i;
```

Comment:
This will not pass strinct signed/unsigned conversion checks, which I'd like to turn on at some point. I think the clearest way to do this with std algorithms is an outer std::find_if and a std::any_of nested inside.